plugins/put-a-file-with-ftp/impl/src/test/java/org/pentaho/di/job/entries/ftpput/JobEntryFTPPUTTest.java

```diff
@@ -89,6 +114,43 @@ public void createFtpClient_UsesTimeoutIfSet() throws Exception {
     verify( ftpClient ).setTimeout( 10 );
   }
 
+  @Test
+  public void testLocalDirDoesntExist() throws Exception{
+    this.initialSetUp();
+    Result result = entry.execute( previousResult, 1 );
+    assertEquals( 1, result.getNrErrors() );
+  }
+
+  @Test
+  public void testNoFilePermissions() throws Exception {
```

Comment:
This test case `testNoFilePermissions()` doesn't make any sense.

 The new code you added throws an exception
`throw new FTPException(...`
But yet here you aren't checking for the thrown FTOException anywhere.

And somehow, the `execute()` return with a result.

Reply:
fixed with https://github.com/pentaho/pentaho-kettle/commit/42a31cf00e0008515d6a51939fcac7cd624e1075